### PR TITLE
Added a deployment for APIM-Product-Subscription

### PR DESCRIPTION
**Pull request by Knyatsine**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -113,11 +119,39 @@
...
+                },
+                "parameters": {
+                    "apimName": {
+                        "value": "[parameters('sharedApimName')]"
+                    },
+                    "subscriptionName": {
+                        "value": "[variables('AppServiceName')]"
+                    },
+                    "subscriptionScope": {
+                        "value": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', parameters('sharedApimResourceGroup'), '/providers/Microsoft.ApiManagement/service/', parameters('sharedApimName'), '/products/FindApprenticeshipTrainingOuterApi')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:142]: Is that the right product?

Comment (Knyatsine) [reply azure/template.json:142]: @jack-cloud-platform  the classic mistake of copy and paste lol

File: azure/template.json [hunk 2]
```diff
@@ -113,11 +119,39 @@
...
+                },
+                "parameters": {
+                    "apimName": {
+                        "value": "[parameters('sharedApimName')]"
+                    },
+                    "subscriptionName": {
+                        "value": "[variables('AppServiceName')]"
+                    },
+                    "subscriptionScope": {
+                        "value": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', parameters('sharedApimResourceGroup'), '/providers/Microsoft.ApiManagement/service/', parameters('sharedApimName'), '/products/EmployerDemandWebOuterApi')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:142]:
```suggestion
                        "value": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', parameters('sharedApimResourceGroup'), '/providers/Microsoft.ApiManagement/service/', parameters('sharedApimName'), '/products/EmployerDemandOuterApi')]"
```

File: azure/template.json [hunk 3]
```diff
@@ -113,11 +119,39 @@
...
+            }
+        },
         {
             "type": "Microsoft.Resources/deployments",
             "apiVersion": "2020-06-01",
             "name": "app-service",
             "resourceGroup": "[variables('resourceGroupName')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Resources/deployments', 'apim-product-subscription')]"
+            ],
```
[the remark this answers is not in the document]
Comment (Knyatsine) [reply azure/template.json:154]: It's used in findingapprenticeship template but haven't come across in other ones, so might not be needed

File: azure/template.json [hunk 4]
```diff
@@ -61,7 +67,7 @@
     },
     "resources": [
         {
-            "apiVersion": "2020-06-01",
+            "apiVersion": "2020-10-01",
```
Comment (jack-cloud-platform) [inline azure/template.json:70]: Is this version supported?

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**